kerosene/nn/apex.py

```diff
@@ -56,39 +58,95 @@ def backward(self, gradient: Optional[Tensor] = None, retain_graph=False,
             self._loss.backward(gradient, retain_graph, create_graph)
 
     def mean(self):
-        self._loss = torch.mean(self._loss)
-        return self
+        loss = torch.mean(self._loss)
```

Comment:
would def need tests on these operation as they are crucial. :point_down: I  see a potential GC overhead limit error (cause of potential of huge number of object to clear anmd quick) here but we'll see.

Reply:
I see. But it's the only way I can think that doesn't change the value after in user's space. It's the only way I can thing to get this correct behaviour.